File: golang/22.08/66.go
func plusOne(digits []int) []int {
    var ans []int
    ch := false
    
    for i := len(digits) - 1; i >= 0; i-- {
        digits[i] = (1 + digits[i]) % 10
        if digits[i] != 0 {
            ch = true
            break
        }
    }
    
    a := []int{1}
    
    if ch == false {
        ans = append(a, digits...)
    } else {
        ans = digits
    }
    
    
    return ans
}

File: javascript/23.06/66.py
function plusOne(digits: number[]): number[] {
    let ch = false
    for (let i = digits.length-1; i > -1; i--) {
        if (digits[i] != 9) {
            digits[i]++
            break
        } else {
            digits[i] = 0
            if (i == 0) {
                ch = true
            }
        }
    }

    if (ch) {
        digits = [1].concat(digits)
    }

    return digits
};
